app/Services/Phoenix/Phoenix.php

```diff
@@ -55,4 +60,55 @@ public function getReportback($reportback_id, $reportback_item_id)
     {
         return $this->get('reportbacks/' . $reportback_id);
     }
+
+    /**
+     * Get an index of (optionally filtered) campaign signups from Phoenix.
+     * @see: https://github.com/DoSomething/phoenix/wiki/API#retrieve-a-signup-collection
+     *
+     * @param array $query - query string, for filtering results
+     * @return array - JSON response
+     */
+    public function getAllSignups(array $query = [])
+    {
+        $path = 'signups';
+
+        // Avoid caching when requesting signups for sepcific user and campaign.
+        // @Temporary?
+        if (isset($query['campaigns']) && isset($query['users'])) {
+            return $this->get($path, $query);
+        }
+
+        return $this->get($path, $query);
+    }
+
+    /**
+     * Get details for a particular campaign signup from Phoenix.
+     * @see: https://github.com/DoSomething/phoenix/wiki/API#retrieve-a-specific-signup
+     *
+     * @return array - JSON response
+     */
+    public function getSignup($signup_id)
+    {
+        $path = 'signups/'.$signup_id;
+
+        return $this->get($path);
+    }
+
+    /**
+     * Get an index of (optionally filtered) campaign reportbacks from Phoenix.
+     * @see: https://github.com/DoSomething/phoenix/wiki/API#retrieve-a-reportback-collection
+     *
+     * @param array|string $query - query string, for filtering results
+     * @return array - JSON response
+     */
+    public function getAllReportbacks(array $query = [])
+    {
+        $path = 'reportbacks';
+        $query['load_user'] = true;
+        if (auth()->id()) {
+            $query['as_user'] = auth()->id();
```

Comment:
@weerd I'm not sure what this is for. honestly. It looks like yes, it will be the currently signed in user which should be fine since only admins are allowed to log in. But also we aren't really using this method.